wallet/pskt/src/wasm/pskt.rs

```diff
@@ -384,4 +396,296 @@ impl PSKT {
+                                pub_key: keypair.public_key(),
+                                key_source: None,
+                            })
+                        })
+                    })
+                })
+                .collect::<Result<Vec<_>>>()?;
+
+            Ok(signatures)
+        })?;
+
+        self.replace(State::Signer(signed_pskt))
+    }
+
+    /// Get `Transaction` from a PSKT, by first finalizing it.
+    /// This method is useful to broadcast PSKT to the Kaspa Network, using `RpcClient.submitTransaction`.
+    #[wasm_bindgen(js_name = "finalizeAndExtractTransaction")]
+    pub fn finalize_and_extract_transaction(&self, network_type: &NetworkTypeT) -> Result<Transaction> {
+        let network_type: NetworkType = network_type.try_into()?;
+
+        let pskt_finalizer: Native<Finalizer> = match self.take() {
+            State::NoOp(inner) => inner.ok_or(Error::NotInitialized)?.into(),
+            State::Creator(pskt) => pskt.constructor().signer().finalizer(),
+            State::Constructor(pskt) => pskt.signer().finalizer(),
+            State::Updater(pskt) => pskt.signer().finalizer(),
+            State::Signer(pskt) => pskt.finalizer(),
+            State::Combiner(pskt) => pskt.signer().finalizer(),
+            State::Finalizer(pskt) => pskt,
+            state => return Err(Error::state(state))?,
+        };
+
+        let result = pskt_finalizer.finalize_sync(|inner: &Inner| -> Result<Vec<Vec<u8>>> {
+            Ok(inner
+                .inputs
+                .iter()
+                .map(|input| -> Vec<u8> {
+                    let signatures: Vec<_> = input
+                        .partial_sigs
+                        .clone()
+                        .into_iter()
+                        .flat_map(|(_, signature)| {
+                            iter::once(OpData65).chain(signature.into_bytes()).chain([input.sighash_type.to_u8()])
+                        })
+                        .collect();
+
+                    signatures
+                        .into_iter()
+                        .chain(
+                            input
+                                .redeem_script
+                                .as_ref()
+                                .map(|redeem_script| ScriptBuilder::new().add_data(redeem_script.as_slice()).unwrap().drain().to_vec())
+                                .unwrap_or_default(),
+                        )
+                        .collect()
+                })
+                .collect())
+        });
+
+        let finalized_pskt = match result {
+            Ok(finalized_pskt) => finalized_pskt,
+            Err(e) => return Err(Error::from(e.to_string())),
+        };
+
+        let mutable_transaction = finalized_pskt
+            .extractor()?
+            .extract_tx(&network_type.into())
+            .map_err(|e| Error::custom(format!("Failed to extract transaction: {e}")))?;
+
+        Ok(mutable_transaction.tx.into())
```

Comment:
Is there a better interface to expose here, so end-user would still be able to modify priority_fees and other parameters of their choices?